File: java/src/main/java/com/meteroid/models/CheckoutSessionStatus.java
// This file is @generated
package com.meteroid.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.meteroid.Utils.ToQueryParam;

public enum CheckoutSessionStatus implements ToQueryParam {
    CREATED("CREATED"),
    AWAITING_PAYMENT("AWAITING_PAYMENT"),
    COMPLETED("COMPLETED"),
    EXPIRED("EXPIRED"),
    CANCELLED("CANCELLED");
    private final String value;

    CheckoutSessionStatus(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: python/meteroid/models/checkout_session_status.py
# this file is @generated
import enum


class CheckoutSessionStatus(str, enum.Enum):
    CREATED = "CREATED"
    AWAITING_PAYMENT = "AWAITING_PAYMENT"
    COMPLETED = "COMPLETED"
    EXPIRED = "EXPIRED"
    CANCELLED = "CANCELLED"

    def __str__(self) -> str:
        return str(self.value)
